lib/shelley/test/integration/Main.hs

```diff
@@ -0,0 +1,220 @@
+    ( ByronKey )
+import Cardano.Wallet.Primitive.AddressDerivation.Icarus
+    ( IcarusKey )
+import Cardano.Wallet.Primitive.Types
+    ( Address (..)
+    , FeePolicy (..)
+    , GenesisBlockParameters (..)
+    , Hash (..)
+    , SyncTolerance (..)
+    , TxIn (..)
+    , TxOut (..)
+    , TxParameters (..)
+    )
+import Cardano.Wallet.Shelley
+    ( SomeNetworkDiscriminant (..)
+    , serveWallet
+    , setupTracers
+    , tracerSeverities
+    )
+import Cardano.Wallet.Shelley.Compatibility
+    ( Shelley )
+import Cardano.Wallet.Shelley.Faucet
+    ( initFaucet )
+import Cardano.Wallet.Shelley.Launch
+    ( withCardanoNode )
+import Cardano.Wallet.Shelley.Transaction.Size
+    ( maxSizeOf, minSizeOf, sizeOfSignedTx )
+import Control.Concurrent.Async
+    ( race )
+import Control.Concurrent.MVar
+    ( newEmptyMVar, putMVar, takeMVar )
+import Control.Exception
+    ( throwIO )
+import Control.Monad
+    ( forM_, void )
+import Data.Proxy
+    ( Proxy (..) )
+import Data.Quantity
+    ( Quantity (..) )
+import Data.Text
+    ( Text )
+import Network.HTTP.Client
+    ( defaultManagerSettings
+    , managerResponseTimeout
+    , newManager
+    , responseTimeoutMicro
+    )
+import Numeric.Natural
+    ( Natural )
+import System.IO
+    ( BufferMode (..), hSetBuffering, stdout )
+import System.IO.Temp
+    ( withSystemTempDirectory )
+import Test.Hspec
+    ( Spec, SpecWith, after, describe, hspec, parallel )
+import Test.Hspec.Extra
+    ( aroundAll )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , Headers (..)
+    , KnownCommand (..)
+    , Payload (..)
+    , TxDescription (..)
+    , request
+    , unsafeRequest
+    )
+import Test.Utils.Paths
+    ( getTestData )
+
+import qualified Cardano.Wallet.Api.Link as Link
+import qualified Data.Aeson as Aeson
+import qualified Data.ByteString as BS
+import qualified Data.Text as T
+import qualified Test.Integration.Scenario.API.Network as Network
+import qualified Test.Integration.Scenario.CLI.Keys as KeyCLI
+import qualified Test.Integration.Scenario.CLI.Miscellaneous as MiscellaneousCLI
+import qualified Test.Integration.Scenario.CLI.Mnemonics as MnemonicsCLI
+import qualified Test.Integration.Scenario.CLI.Network as NetworkCLI
+import qualified Test.Integration.Scenario.CLI.Port as PortCLI
+
+-- | Define the actual executable name for the bridge CLI
+instance KnownCommand Shelley where
+    commandName = "cardano-wallet-shelley"
+
+main :: forall t . t ~ Shelley => IO ()
+main = withUtf8Encoding $ withLogging Nothing Info $ \(_, tr) -> do
+    hSetBuffering stdout LineBuffering
+    hspec $ do
+        describe "No backend required" $ do
+            describe "Mnemonics CLI tests" $ parallel (MnemonicsCLI.spec @t)
+            describe "Miscellaneous CLI tests" $ parallel (MiscellaneousCLI.spec @t)
+            describe "Key CLI tests" $ parallel (KeyCLI.spec @t)
+        describe "API Specifications" $ specWithServer tr $ do
+            Network.spec
```

Comment:
What about the wallet specs?

Reply:
They need transactions. For faucet txs I think.

Reply:
> For faucet txs I think.

Precisely not :) The faucets are genesis UTxOs, so most of the wallet scenarios are actually about restoring wallets from on-chain data.

Reply:
Ok. The node has stopped producing blocks for some reason 🤔

Reply:
There are a few scenarios requiring transactions. I can't mark them pending since they exist in core-integration. Let's deal with this later.